### ISSUE TITLE
CLI scripts print 'Application Instantiation Error' when redis cache enabled

### ISSUE THREAD

**Issue by csthomas** [joomla/joomla-cms#16903]
### Steps to reproduce the issue
Install joomla with cache Redis enabled.

Go to joomla folder and run `php -f cli/update_cron.php`

I get result:
```
Error displaying the error page: Application Instantiation Error: Application Instantiation Error
```

### Expected result
Script works without errors

### Actual result
Get exception message.

### System information (as much as possible)
PHP 7.0 and 7.1 - not revelant
Joomla 3.7.3-dev and olders
Redis as cache handler and System Cache ON

### Additional comments
The error comes from https://github.com/joomla/joomla-cms/blob/staging/libraries/joomla/factory.php#L122

and it starts at https://github.com/joomla/joomla-cms/blob/staging/libraries/joomla/cache/storage/redis.php#L67

**Can we break B/C by removing `$app = JFactory::getApplication();` and replacing
`JError::raiseWarning(500, 'Redis connection failed');`
by `throw new JCacheExceptionConnecting('Could not connect to redis server');`
 in redis handler?**

If yes then I can write PR.

**Comment by mbabker** [comment]
It's been argued adding new Exceptions to a method is a B/C break.  If it's already throwing Exceptions, we're fine.  Otherwise, I honestly don't want to get into the politics of it all again.

**Comment by csthomas** [comment]
I put my solution at #16904

**Comment by franz-wohlkoenig** [comment]
closed as having PR #16904 <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16903">issues.joomla.org/tracker/joomla-cms/16903</a>.</sub>
